### ISSUE TITLE
source-host nor sender-host work for nsca on nsclient++ 0.5.0062

### ISSUE THREAD

**Issue by Dearl1988** [mickem/nscp#510]
I send a passive check for our nagios core to alert whether a service works or not... You see, the server hostname is capitalized and the nagios host is in lowercase letters, the line of powershell code below used to work just fine on nsclient++ 0.4.2.114

```
`.\nscp.exe client --module NSCAClient address=serverIP:portNumber encryption=XOR **source-host="hostnameServer"** command=service result=statusNumber message=statusInformation`
```

### Expected Behavior

Nagios core should accept that passive check and tell us what server the check is coming from

### Actual Behavior

It seems to be a conflict in what nagios is supposed to accept as a hostname and when I send my passive check, nagios event log tells me there is not host with that name.

### Details

* NSClient++ version: 0.5.0062
* OS and Version: Windows Server 2016 Datacenter
* Checking from: Centos
* Checking with: check_nsca

### Additional Details

NSClient++ log:
```
[28-11-2017 14:00:37] Warning: Passive check result was received for service 'MDI_zip_d' on host 'MDIFS24', but the host could not be found!
```

**Comment by box293** [comment]
Nagios is case sensative. NSClient is sending the hostname as it's been typed by your admin who setup that PC. 

To resolve such problems try changing how the hostname is sent:
http://docs.nsclient.org/reference/client/NSCAClient/#/settings/NSCA/client_hostname
HOSTNAME

The host name of the monitored computer. Set this to auto (default) to use the windows name of the computer.
```
auto Hostname 
${host} Hostname 
${host_lc} Hostname in lowercase 
${host_uc} Hostname in uppercase 
${domain} Domainname 
${domain_lc} Domainname in lowercase 
${domain_uc} Domainname in uppercase
```

**Comment by Dearl1988** [comment]
You're right about Nagios being case sensitive, but by knowing that and the object on Nagios being configured in lower case, and the pc hostname in upper case, I wanna do that by sending the hostname matching the object using the variable **host-sender** just like I used to do in previous version... But, it seems like nagios ain't accepting that variable, it's almost like it doesn't recognize it.

**Comment by box293** [comment]
What was the previous version?

**Comment by Dearl1988** [comment]
Sorry for the delay... I was out on paternity leave... The previous version
was 0.4.2.114 and now it is 0.50062...

I don't wanna tweak with the .ini file

This is the one liner


nscp client --module NSCAClient address=nagiosServer:5667*
source-host=mdifs23 *encryption=XOR command="mdiFiles_cleanup_d" result=2
message=chaos"

2017-11-30 19:26 GMT-02:00 Troy Lea <notifications@github.com>:

**Comment by mickem** [comment]
Testing on 0.5.2 I get the following:

```
c:\source\build\x64\dev
$ nscp client --log trace --module NSCAClient address=nagiosServer:5667* source-host=mdifs23 encryption=XOR command="mdiFiles_cleanup_d" result=2 message=chaos"
...
T       nsca Target configuration: address: ://nagiosServer:5667*, timeout: 30, retry: 2, data: { delay: 0, encoding: , encryption: XOR, password: , payload length: 512, retries: 3, time offset: 0, }
T       nsca Scheduling packet: host: mdifs23, service: mdiFiles_cleanup_d, code: 3, time: 1515442060, result: chaos
T       nsca Connecting to: host: nagiosServer:5667, buffer_length: 512, time_delta: 0, password: , encryption: XOR(1), hostname: mdifs23, encoding: , ssl: ssl disabled
D       core Attempting to stop all plugins
Socket error: resolve: Ingen sådan värd är kändD       core Stopping all plugins

D       core Unloading plugin: NSCAClient...
D       core Stopping: COM helper
D       core Stopping: Settings instance
```

Which seems right unless I am misstaken...

**Comment by mickem** [comment]
Please reopen if you still have issues in 0.5.2

**Comment by Dearl1988** [comment]
I still have issues

**Comment by mickem** [comment]
What issues do you have? same or?
If I use the above example it works for me so please show what you get with the above command...

**Comment by mickem** [comment]
Likely fixed in https://github.com/mickem/nscp/releases/tag/0.12.1 (atleast I fixed an issue with hostname in that release)
